main.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'app'
+
+puts 'Welcome to your school library'
+
```

Comment:
***Since some of your app files do not have commits and do not appear in this pull request, I will use the `main.rb` file to comment on other files in your app.***

1. To follow the best practices in Ruby and fix the following errors. Please use `require_relative` in all your files since all of the are in the same directory.

![image](https://user-images.githubusercontent.com/38088818/210895338-4a701746-107d-4138-8e1b-9e90a99e348e.png)

- To fix this error please use `require_relative` in the `base_decorator.rb` file.

2. You misspelled `attr_accessor` in your `BaseDecorator` class which is causing the following error. Kindly look into it.

![image](https://user-images.githubusercontent.com/38088818/210896194-a2821e8b-57da-4fd5-9b3a-77a55b1721a0.png)

3. You have an error pertaining to your `person.rb` file.

![image](https://user-images.githubusercontent.com/38088818/210897185-0cae9f2e-80e5-4d48-9b75-29504104c628.png)

- To remove it, please remove the following test code in your file.

![image](https://user-images.githubusercontent.com/38088818/210897436-d90862d8-eadb-4553-a2cc-74513d01bc11.png)

4. There is no method defined as `start` in your `app.rb` file. Which results in this error

![image](https://user-images.githubusercontent.com/38088818/210897691-fbc1b803-b74b-45c1-a9ee-28d7e2339618.png)

![image](https://user-images.githubusercontent.com/38088818/210897878-4a64540c-1f1b-41a9-ab76-a8487c021231.png)

- Try `app.run` instead...

5. [OPTIONAL] Since you already have the greeting `Welcome to School Library App!` in the method `run` in `app.rb` , I think it would be great to remove it from the `main.rb` line 3 to prevent this duplication.

![image](https://user-images.githubusercontent.com/38088818/210898502-bfde4902-7516-4812-9553-e81ade847aec.png)

Reply:
@KDlamini Thank you for your kind review